x-pack/plugins/code/public/sagas/structure.ts

```diff
@@ -18,15 +24,33 @@ function requestStructure(uri?: string) {
   });
 }
 
-function* handleLoadStructure(action: Action<string>) {
+function* beginPollingSymbols(action: Action<string>) {
   try {
-    const data = yield call(requestStructure, `git:/${action.payload}`);
-    yield put(loadStructureSuccess({ path: action.payload!, data }));
+    const pollingTaskId = yield fork(pollingSaga, action);
+    yield take([String(loadStructureSuccess), String(loadStructureFailed)]);
+    yield cancel(pollingTaskId);
   } catch (err) {
     yield put(loadStructureFailed(err));
   }
 }
 
 export function* watchLoadStructure() {
-  yield takeEvery(String(loadStructure), handleLoadStructure);
+  yield takeEvery(String(loadStructure), beginPollingSymbols);
+}
+
+function* pollingSaga(action: Action<string>) {
+  while (true) {
+    try {
+      const data = yield call(requestStructure, `git:/${action.payload}`);
+      yield put(loadStructureSuccess({ path: action.payload!, data }));
+      yield delay(POLLING_INTERVAL * 1000);
```

Comment:
`break` here and remove the `delay`, you don't need to continue the loop anymore.